app/controllers/books_controller.rb

```diff
@@ -59,9 +59,9 @@ def destroy
 
   private
 
-  # Use callbacks to share common setup or constraints between actions.
   def set_book
-    @book = Book.find(params[:id])
+    @book = Book.includes(comments: :user).find(params[:id])
+    @book.comments = @book.comments.order(:created_at, :id)
```

Comment:
あ、この操作は非常に危険ですね。たとえば

```suggestion
    @book.comments = []
```

みたいな代入をすると`@book`に紐付くコメントが全部削除されます。
すなわち、この代入によってDBの追加 or 削除が発生します。

そもそも63行目でincludesするとorderを自由に決められないので、いったんBookをfindするだけにしましょう。

```suggestion
    @book = Book.find(params[:id])
```

Reply:
> この代入によってDBの追加 or 削除が発生します。

rails console から確認しました。インスタンス変数に値をセットするだけでDBが更新されるんですね..💦 

> includesするとorderを自由に決められないので、いったんBookをfindするだけにしましょう

コメントありがとうございます。show.html.erbでorderメソッドを指定するように修正しました！

Reply:
> インスタンス変数に値をセットするだけでDBが更新されるんですね..💦

厳密にいえば「インスタンス変数に値をセットする」とは限らないですね。
Rubyはクラスごとに`=`の挙動を自由に定義できるので。（チェリー本 7.10.4参照）

この場合はRailsが`=`に対して意図的にデータを更新する挙動を割り当てている、ということになります。

> joinモデルのコレクションは、[has_many関連付けメソッド](https://railsguides.jp/association_basics.html#has-many%E9%96%A2%E9%80%A3%E4%BB%98%E3%81%91)経由で管理できます。たとえば、以下のような割り当てを実行したとします。
> 
> `physician.patients = patients`
>
> このとき、新たに関連付けられたオブジェクトについて、新しいjoinモデルが自動的に作成されます。以前あった行がなくなった場合は、その行はjoinモデルから自動的に削除され、joinモデルに含まれなくなります。
>
> https://railsguides.jp/association_basics.html#has-many-through%E9%96%A2%E9%80%A3%E4%BB%98%E3%81%91

Reply:
ご説明ありがとうございます！
今回は `has_many` で関連付けた場合の `=` の挙動なんですね。